### PR TITLE
feat: add megaphone api token auth

### LINKED ISSUE #1164: Add authentication token for reading megaphone API
Megaphone API uses a basic bearer token in the Authentication header. The connection node needs to include this from config to access the API.

### PR DESCRIPTION
**Pull request by bbangert**
Closes #1164

### CODE REVIEW

File: autopush/main_argparse.py [hunk 1]
```diff
@@ -231,6 +231,9 @@ def parse_connection(config_files, args):
     parser.add_argument('--megaphone_api_url',
                         help="The megaphone API URL to query for updates",
                         default=None, type=str, env_var="MEGAPHONE_API_URL")
+    parser.add_argument('--megaphone_api_token',
+                        help="The megaphone API URL to query for updates",
```
Comment (pjenvey) [inline autopush/main_argparse.py]: copy pasta

File: autopush_rs/src/server/mod.rs [hunk 2]
```diff
@@ -326,7 +330,10 @@ impl Server {
     fn new(opts: &Arc<ServerOptions>, tx: queue::Sender) -> Result<(Rc<Server>, Core)> {
         let core = Core::new()?;
         let broadcaster = if let Some(ref megaphone_url) = opts.megaphone_api_url {
-            ServiceChangeTracker::with_api_services(megaphone_url)
+            let megaphone_token = opts.megaphone_api_token.as_ref().expect(
+                "Megaphone API requires a Megaphone API Token to be set".into()
```
Comment (pjenvey) [inline autopush_rs/src/server/mod.rs]: nitpick .into() isn't necessary in these 2 expects. I guess we have them elsewhere but they aren't needed

File: autopush/tests/test_rs_integration.py [hunk 3]
```diff
@@ -69,9 +69,11 @@ class MockMegaphoneRequestHandler(BaseHTTPRequestHandler):
     API_PATTERN = re.compile(r'/v1/broadcasts')
     services = {}
     polled = Event()
+    token = uuid.uuid4().hex
```
Comment (pjenvey) [inline autopush/tests/test_rs_integration.py]: Note that the auth header should be "Bearer sometoken". This patch requires "Bearer " to be specified along side the token in the config.

megaphone sort of has to hardcode "Bearer" in its code I think. I'd lean towards these 2 configs matching. If not, at the very least I'd call this config value a "header" vs a "token" and document it w/ an example. and probably include a "Bearer " on this line as well

(autopush_connection.ini could also probably use an example)

**Comment by jrconlin** [comment]
Well, if you want to get SUPER Picky, the schema + token bit are "credentials" according to the RFCs. 

Please don't be super picky.

**Pull request merged**